src/main/java/org/haffson/adventofcode/service/AdventOfCodeService.java

```diff
@@ -43,13 +43,18 @@ public AdventOfCodeService(List<Days> daysSolutions) {
      * @return a {@code String} with the result for the puzzle, or in case it has not been implemented,
      * an {@link PuzzleNotSolvedYetException} is thrown.
      */
-    public String getResultsForASpecificDayAndPuzzlePart(String day, String part) {
-        Days thisDaysClass = findDayForDay(Integer.parseInt(day));
+    public String getResultsForASpecificDayAndPuzzlePart(Integer day, Integer part) {
+
+        Objects.requireNonNull(day, "day is null");
+        Objects.requireNonNull(part, "part is null");
+
+
+        Days thisDaysClass = findDayForDay(day);
         if (!isProblemSolvedForPart(thisDaysClass, part)) {
             throw new PuzzleNotSolvedYetException(new Throwable());
-        } else if (("1").equals(part)) {
+        } else if ((part).equals(1)) {
```

Comment:
Alt+Enter : "remove unnecessary parentheses"

Alt+Enter : "flip equals" -> momentan ist das nicht null-safe, falls part null ist.

Reply:
Ich habe in der Methode folgendes hinzugefügt:
Objects.requireNonNull(day, "day is null");
Objects.requireNonNull(part, "part is null");

Zusätzlich funktioniert equals() nicht mit int. Und 1 und 2 sind ja int, weswegen ein flip nur möglich wäre, wenn ich diese zu Integers umwandel. Sollte man das tun, oder es dann so lassen? Auch "==" würde ja jetzt funktionieren, benutzt man dann lieber == oder equals()?

Reply:
Bei primitiven Typen wie `ìnt` lieber `==`.

Ein bisschen aufpassen musst du, wenn einer der Operanden doch ein Objekt ist (zB. `Integer`): Dann führt die Belegung mit `null` zu einer Exception:

    Integer a = null;
    var eq = a == 1; // Exception

Mit `Objects.equals()` muss man sich über solche Details keine Gedanken machen:

    Integer a = null;
    var eq = Objects.equals(a, 1);